File: 3516.find-closest-person/solution.go
func findClosest(x int, y int, z int) int {
    switch d := abs(x - z) - abs(y - z); {
        case d < 0: return 1
        case d > 0: return 2
        default: return 0
    }
}

func abs(a int) int {
    if a < 0 {
        return -a
    }
    return a
}

File: 3516.find-closest-person/solution.kt
class Solution {
    fun findClosest(x: Int, y: Int, z: Int): Int {
        val dx = abs(x - z)
        val dy = abs(y - z)
        return when {
            dx < dy -> 1
            dx > dy -> 2
            else -> 0
        }
    }
}